crates/wasi-common/src/sys/windows/hostcalls_impl/fs.rs

```diff
@@ -120,9 +135,22 @@ pub(crate) fn path_open(
 ) -> Result<File> {
     use winx::file::{AccessMode, CreationDisposition, Flags};
 
+    let is_trunc = oflags & wasi::__WASI_OFLAGS_TRUNC != 0;
+
+    if is_trunc {
+        // Windows requires write access for truncation
+        if !write {
+            return Err(Error::ENOTCAPABLE);
```

Comment:
This shouldn't be addressed in the platform-specific code. On Unix the behavior `O_TRUNC` + `O_RDONLY` is undefined too. Therefore, as discussed in #557, `path_open` should probably be called with `write = true`, when the file is to be truncated, as `RIGHT_PATH_FILESTAT_SET_SIZE` is the required permission for truncation. But then, analogically to what was brought up in #618, the file descriptor will spuriously get write permissions.

We probably should decide how to resolve #618 first.

Reply:
Agreed, some resolution to those issues to clarify how truncation should work would be really helpful :)

I'll remove the checks once we figure that out.